src/backend/deep_search/rag.py

```diff
@@ -0,0 +1,586 @@
+            return
+
+        # Create new collection
+        logger.info(f"Creating new collection: {self.collection_name}")
+        response = self.session.post(
+            f"{self.config.base_url}/collections",
+            json={"name": self.collection_name, "model": self.embeddings_model}
+        )
+        response.raise_for_status()
+        self.collection_id = response.json()["id"]
+        logger.info(f"Collection '{self.collection_name}' created with ID: {self.collection_id}")
+
+    def clear_collection(self):
+        """Clear all documents from the current collection."""
+        if not self.collection_id:
+            return
+            
+        try:
+            # Clear the collection by deleting it and recreating it
+            # This is often easier than trying to delete individual documents
+            response = self.session.delete(f"{self.config.base_url}/collections/{self.collection_id}")
+            response.raise_for_status()
+            logger.info(f"Deleted collection {self.collection_id}")
+            
+            # Reset state
+            self.collection_id = None
+            self.indexed_email_ids = set()
+            self.last_indexed_hash = None
+            self.last_index_time = None
+            
+            # Recreate the collection
+            self.create_collection()
+            
+        except Exception as e:
+            logger.error(f"Error clearing collection: {e}")
+            # Reset state anyway
+            self.collection_id = None
+            self.indexed_email_ids = set()
+            self.last_indexed_hash = None
+            self.last_index_time = None
+
+    def add_single_email(self, email: dict):
+        """
+        Add a single email to the collection without triggering full reindexing.
+        This is efficient for real-time updates when new messages arrive.
+        
+        Args:
+            email: Dictionary with 'id', 'body', and other email fields
+        """
+        if not self.collection_id:
+            logger.warning("Cannot add single email: no collection exists")
+            return False
+            
+        email_id = email.get('id')
+        if not email_id:
+            logger.error("Cannot add email without ID")
+            return False
+            
+        # Check if already indexed
+        if email_id in self.indexed_email_ids:
+            logger.debug(f"Email {email_id} already indexed, skipping")
+            return True
+            
+        try:
+            logger.info(f"Adding single email {email_id} to collection {self.collection_id}")
+            
+            # Prepare document for indexing
+            documents = [{
+                "id": email_id,
+                "content": email.get('body', ''),
+                "metadata": {
+                    "subject": email.get('subject', ''),
+                    "sender": email.get('sender', ''),
```

Comment:
It could interesting to index recipients nope ? (to and cc)
Maybe also if email has attachments just through a boolean value ?